xcodeproj/internal/top_level_targets.bzl

```diff
@@ -455,7 +459,7 @@ def process_top_level_target(
     if not ctx.var["COMPILATION_MODE"] == "dbg":
         build_settings["DEBUG_INFORMATION_FORMAT"] = ""
     elif not cpp.apple_generate_dsym:
-        build_settings["DEBUG_INFORMATION_FORMAT"] = "dwarf"
+        build_settings["DEBUG_INFORMATION_FORMAT"] = "dwarf-with-dsym"
```

Comment:
Not sure if it's worth changing, I'll check it out.

Reply:
`not cpp.apple_generate_dsym`, so we need an `else` case here to account for `dwarf-with-dsym`.

Reply:
Actually, I didn't do that, because that is the default. Did you find yourself needing this change at all?

Reply:
According to [this](https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html) documentation, the default is `dwarf`.
I don't notice any issues with this value, but with the current state in the master, `DEBUG_INFORMATION_FORMAT` value just an empty string, which just looks confusing if `apple_generate_dsym` is enabled.

So I think it make sense to change it to:
```
elif cpp.apple_generate_dsym:
        build_settings["DEBUG_INFORMATION_FORMAT"] = "dwarf-with-dsym"
```

I quickly checked and it looks like it should be this way.

Reply:
In `CreateProject` in the generator let's set `DEBUG_INFORMATION_FORMAT = "dwarf"` to make this explicit then.